core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -236,6 +236,7 @@ class LogCleaner(val config: CleanerConfig,
           try {
             endOffset = cleaner.clean(cleanable)
             recordStats(cleaner.id, cleanable.log.name, cleanable.firstDirtyOffset, endOffset, cleaner.stats)
+            cleanerManager.decrementNumFilthyLogs()
```

Comment:
Thanks a lot for the feedback @junrao 

Please see my comment below. We would want to decrement "num-filthy-logs" only if the cleaner  succeeded in cleaning the log (Line 237).  Moving it to finally block could decrement the counter even if the cleaner failed.  Since we want to track if the cleaner failed, it's added to the end of the try block.